File: app/src/main/java/ch/unstable/ost/database/dao/FavoriteConnectionDao.java
package ch.unstable.ost.database.dao;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ch.unstable.ost.database.model.FavoriteConnection;
import io.reactivex.Flowable;
import io.reactivex.Single;

@Dao
public interface FavoriteConnectionDao {
    @Query("SELECT * FROM " + FavoriteConnection.TABLE_NAME + " ORDER BY creation_date DESC LIMIT 100")
    Flowable<List<FavoriteConnection>> getFavoriteConnections();

    @Insert
    long addConnection(FavoriteConnection connections);

    @Query("SELECT * FROM " + FavoriteConnection.TABLE_NAME + " ORDER BY creation_date DESC LIMIT 1")
    Single<FavoriteConnection> getLatestFavorite();

    @Delete
    void removeConnectionById(FavoriteConnection id);

    @Query("SELECT * FROM " + FavoriteConnection.TABLE_NAME + " WHERE id = :id")
    Single<FavoriteConnection> getFavoriteById(long id);
}

File: app/src/main/java/ch/unstable/ost/database/dao/FavoriteConnectionDao.kt
package ch.unstable.ost.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import ch.unstable.ost.database.model.FavoriteConnection
import io.reactivex.Flowable
import io.reactivex.Single

@Dao
interface FavoriteConnectionDao {
    @Query("SELECT * FROM " + FavoriteConnection.TABLE_NAME + " ORDER BY creation_date DESC LIMIT 100")
    fun getFavoriteConnections(): Flowable<List<FavoriteConnection>>

    @Insert
    fun addConnection(connections: FavoriteConnection): Long

    @Query("SELECT * FROM " + FavoriteConnection.TABLE_NAME + " ORDER BY creation_date DESC LIMIT 1")
    fun getLatestFavorite(): Single<FavoriteConnection>

    @Delete
    fun removeConnectionById(id: FavoriteConnection)

    @Query("SELECT * FROM " + FavoriteConnection.TABLE_NAME + " WHERE id = :id")
    fun getFavoriteById(id: Long): Single<FavoriteConnection>
}